sdk/cosmosdb/cosmos/src/globalEndpointManager.ts

```diff
@@ -183,6 +187,60 @@ export class GlobalEndpointManager {
     }
   }
 
+  private refreshStaleUnavailableLocations(): void {
+    const now = Date.now();
+    let isReadRequestType: boolean = true;
+    this.updateLocationList(now, isReadRequestType);
+    this.cleanUnavailableLocationList(now, isReadRequestType);
+
+    isReadRequestType = false;
+    this.updateLocationList(now, isReadRequestType);
+    this.cleanUnavailableLocationList(now, isReadRequestType);
+  }
+
+  private updateLocationList(now: number, isReadRequest: boolean) {
```

Comment:
Is having control over the `isReadRequest` field important for the future? Since the method is just called twice with true & false above, why not do both read & write locations unconditionally?

Some comments on these functions would help understand their functionality and whether or not it's valid to call `updateLocationList` with only read or write and not both.

Reply:
resolved